core/src/main/java/org/elasticsearch/action/bulk/Retry.java

```diff
@@ -102,12 +125,15 @@ public BulkResponse withSyncBackoff(Client client, BulkRequest bulkRequest) thro
         private volatile BulkRequest currentBulkRequest;
         private volatile ScheduledFuture<?> scheduledRequestFuture;
 
-        AbstractRetryHandler(Class<? extends Throwable> retryOnThrowable, BackoffPolicy backoffPolicy, Client client, ActionListener<BulkResponse> listener) {
+        RetryHandler(Class<? extends Throwable> retryOnThrowable, BackoffPolicy backoffPolicy,
+                     BiConsumer<BulkRequest, ActionListener<BulkResponse>> consumer, ActionListener<BulkResponse> listener,
+                     Settings settings, BiFunction<TimeValue, Runnable, ScheduledFuture<?>> scheduleFn) {
             this.retryOnThrowable = retryOnThrowable;
             this.backoff = backoffPolicy.iterator();
-            this.client = client;
+            this.consumer = consumer;
             this.listener = listener;
-            this.logger = Loggers.getLogger(getClass(), client.settings());
+            this.logger = Loggers.getLogger(getClass(), settings);
```

Comment:
Retry.withAsyncBackoff is used by `AbstractAsyncBulkByScrollAction`. I left the `Settings` at the retry logic since I did not want to change how logging is done for that class. Instead `Settings` are not required by the BulkProcessor and it just passes `Settings.EMPTY` to the retry level. Is that not correct? I was just hesitant to change code that will impact other pieces.